### DISCUSSION TITLE
Use z14 merged buildings as input

### DISCUSSION THREAD

**Discussion by wipfli** [onthegomap/planetiler#530] [Q&A]
I would like to use building footprints to generate landcover polygons for built-up areas. I think this could be a bit more convenient than `landuse=residential`.

The approach would be the following: I take the z14 buildings and enlarge them by a certain amount of pixels. Then I merge them. This can be achieved basically by commenting out the unbuffering step in `mergeNearbyPolygons`. The result looks like this at z14:

![image](https://user-images.githubusercontent.com/53421382/227301870-0c51c136-cdb3-401b-be49-d06446a6500b.png)

As a next step, I would like to take this geometry as an input and compute the landcover polygons at all zoom levels.

Is it possible to feed the output of `postProcessLayerFeatures` into Planetiler again?

**Comment by erik** [comment]
Tangential to your question, but NaturalEarth's [Urban Areas](https://www.naturalearthdata.com/downloads/10m-cultural-vectors/10m-urban-area/) data set might be useful, depending on which zoom levels you need this data to be available at. The 10m data isn't detailed enough to look nice at higher zooms, but might be good enough up to ~z9. Could work in combination with OSM-derived data

**Reply by wipfli** [reply]
Thanks. I would like to create such a data set exactly myself from building footprints...

**Reply by wipfli** [reply]
The advantage with generating the landcover from building footprints would be that you can go from z0 to z14 and always use the same data. In openmaptiles and also other maps, at some zoom level they have to switch from natural earth to osm landuse = residential etc. While it is not a big deal, this switch always results in a visual discontinuity when zooming in...

**Comment by wipfli** [comment]
If you set `minPixelSize` to zero, I think actually that you can use the building footprint of the current zoom as an input for the buffer - merge operation. So there is maybe no need to use the z14 merge output at other zoom levels...

But what I am currently struggling with is the fuzziness of the merged polygons:

My code produces something like this:

![image](https://user-images.githubusercontent.com/53421382/227573613-c826b68d-712a-4e5d-97a1-ec6529b703a3.png)

https://wipfli.github.io/blur-buildings/#14/43.73367/7.42121

I saw that planetiler can simplify linestrings with `mergeLineStrings`. Is there something similar to simplify polygons?
